File: bmpm-php/BMPMSephardic.php
<?php

class BMPMSephardic extends BMPM {
	function __construct() {
		$this->setType(BMPM::TYPE_SEPHARDIC);
		
		include "sep/approxcommon.php";
		$this->setApproxCommon($approxCommon);
		
		include "sep/languagenames.php";
		$this->setLanguageNames($languages);
		
		include "sep/lang.php";
		$this->setAllLanguagesBitmap($all);
		$this->setLanguageRules($languageRules);

		$rules = array();
		include "sep/rulesany.php";
		include "sep/rulesfrench.php";
		include "sep/ruleshebrew.php";
		include "sep/rulesitalian.php";
		include "sep/rulesportuguese.php";
		include "sep/rulesspanish.php";
		$this->setRules($rules);

		$exact = array();
		include "sep/exactany.php";
		include "sep/exactfrench.php";
		include "sep/exacthebrew.php";
		include "sep/exactitalian.php";
		include "sep/exactportuguese.php";
		include "sep/exactspanish.php";
		$this->setExact($exact);

		$approx = array();
		include "sep/approxany.php";
		include "sep/approxfrench.php";
		include "sep/approxhebrew.php";
		include "sep/approxitalian.php";
		include "sep/approxportuguese.php";
		include "sep/approxspanish.php";
		$this->setApprox($approx);

		return $this;
	}
}

?>

File: bmpm-cpp/BMPMSephardic.cpp
#include "cpp/php2cpp.h"


#ifndef BMPMSEPHARDIC_CPP
#define BMPMSEPHARDIC_CPP

#include "BMPMSephardic.h"

BMPMSephardic::BMPMSephardic()
    {
        this->setType(BMPM::TYPE_SEPHARDIC);
        #include "sep/approxcommon.h"
        this->setApproxCommon(approxCommon);
        #include "sep/exactcommon.h"
        this->setExactCommon(exactCommon);
        #include "sep/hebrewcommon.h"
        this->setHebrewCommon(hebrewCommon);
        #include "sep/languagenames.h"
        this->setLanguageNames(languages);
        #include "sep/lang.h"
        this->setAllLanguagesBitmap(all);
        this->setLanguageRules(languageRules);
        rules = {};
        #include "sep/rulesany.h"
        #include "sep/rulesfrench.h"
        #include "sep/ruleshebrew.h"
        #include "sep/rulesitalian.h"
        #include "sep/rulesportuguese.h"
        #include "sep/rulesspanish.h"
        this->setRules(rules);
        exact = {};
        #include "sep/exactany.h"
        #include "sep/exactfrench.h"
        #include "sep/exacthebrew.h"
        #include "sep/exactitalian.h"
        #include "sep/exactportuguese.h"
        #include "sep/exactspanish.h"
        this->setExact(exact);
        approx = {};
        #include "sep/approxany.h"
        #include "sep/approxfrench.h"
        #include "sep/approxhebrew.h"
        #include "sep/approxitalian.h"
        #include "sep/approxportuguese.h"
        #include "sep/approxspanish.h"
        this->setApprox(approx);
        return this;
    }

#else
#endif